src/platform/iotjs_module_uart-linux-general.inl.h

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef IOTJS_MODULE_UART_LINUX_GENERAL_INL_H
+#define IOTJS_MODULE_UART_LINUX_GENERAL_INL_H
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/signal.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include "module/iotjs_module_uart.h"
+
+void signal_handler_IO(int status);
+
+iotjs_jval_t jthis;
+int fd;
+
+
+#define UART_WORKER_INIT_TEMPLATE                                             \
+  iotjs_uart_reqwrap_t* req_wrap = iotjs_uart_reqwrap_from_request(work_req); \
+  iotjs_uart_reqdata_t* req_data = iotjs_uart_reqwrap_data(req_wrap);
+
+
+int toBaudConstant(int baudRate) {
+  switch (baudRate) {
+    case 0:
+      return B0;
+    case 50:
+      return B50;
+    case 75:
+      return B75;
+    case 110:
+      return B110;
+    case 134:
+      return B134;
+    case 150:
+      return B150;
+    case 200:
+      return B200;
+    case 300:
+      return B300;
+    case 600:
+      return B600;
+    case 1200:
+      return B1200;
+    case 1800:
+      return B1800;
+    case 2400:
+      return B2400;
+    case 4800:
+      return B4800;
+    case 9600:
+      return B9600;
+    case 19200:
+      return B19200;
+    case 38400:
+      return B38400;
+    case 57600:
+      return B57600;
+    case 115200:
+      return B115200;
+    case 230400:
+      return B230400;
+  }
+  return -1;
+}
+
+
+int toDataBitsConstant(int dataBits) {
+  switch (dataBits) {
+    case 8:
+      return CS8;
+    case 7:
+      return CS7;
+    case 6:
+      return CS6;
+    case 5:
+      return CS5;
+  }
+  return -1;
+}
+
+
+void OpenWorkerUart(uv_work_t* work_req) {
+  UART_WORKER_INIT_TEMPLATE;
+
+  jthis = req_data->jthis;
+  fd = open(iotjs_string_data(&req_data->path), O_RDWR | O_NOCTTY | O_NDELAY);
+
+  if (fd == -1) {
+    req_data->error = kUartErrOpen;
+  } else {
+    req_data->error = kUartErrOk;
+  }
+
+  struct sigaction saio;
+  saio.sa_handler = signal_handler_IO;
+  saio.sa_flags = 0;
+  saio.sa_restorer = NULL;
+  sigaction(SIGIO, &saio, NULL);
+
+  fcntl(fd, F_SETFL, FNDELAY);
+  fcntl(fd, F_SETOWN, getpid());
+  fcntl(fd, F_SETFL, O_ASYNC);
+
+  struct termios options;
+  tcgetattr(fd, &options);
+  options.c_cflag = CLOCAL | CREAD;
+  options.c_cflag |= toBaudConstant(req_data->baudRate);
+  options.c_cflag |= toDataBitsConstant(req_data->dataBits);
+  options.c_iflag = IGNPAR;
+  options.c_oflag = 0;
+  options.c_lflag = 0;
+  tcflush(fd, TCIFLUSH);
+  tcsetattr(fd, TCSANOW, &options);
+}
+
+
+void signal_handler_IO(int status) {
+  sleep(2);
```

Comment:
This signal_handler function is called by both of 'write' function and 'read' function. So I found out it makes a bug somewhere. I intended to bypass the bug with sleep function. But now, I'm planning to fix the bug with using uv_poll() instead of using signal handler directly.